plugin/autosource.vim

```diff
@@ -76,64 +76,44 @@ function! s:CheckHash(path)
     return 1
 endfunction
 
-" Source all `.vimrc` files in your pwd and parents up to your home dir
-function! SourceParentRCs(path, uuid)
-    " Prevent this function from being invoked by multiple autocmds
-    if exists('g:autosource_running') && g:autosource_running !=# a:uuid
-        return
-    elseif has('macunix')
-        if !executable('uuid')
-            echo 'Please install the `uuid` command'
-        endif
-        let g:autosource_running = system('uuid')
-    elseif has('unix')
-        if !executable('uuidgen')
-            echo 'Please install the `uuidgen` command'
-        endif
-        let g:autosource_running = system('uuidgen')
-    endif
-
-    if a:path ==# ''
-        let cur = expand('%:p:h')
-    else
-        let cur = a:path
-    endif
-
-    " don't source outside of home dir
-    if cur !~ $HOME
-        return
-    endif
+let s:fnames = ['.vimrc', '.vimrc.lua']
 
-    " source file if found
-    let rc_exists = 1
-    let rc = cur . '/.vimrc'
-    if !filereadable(rc)
-        let rc = rc . '.lua'
-        if !filereadable(rc)
-            let rc_exists = 0
+" Source all `.vimrc` files in your pwd and parents up to your home dir
+function! s:Source(dir)
+    let cur = a:dir
+    let prev = ''
+    while prev !=# cur
+        " don't source outside of home dir
+        if cur !~ $HOME
+            return
         endif
-    endif
 
-    if rc_exists
-        if s:CheckHash(rc) ==# 1
-            if rc =~ ".lua$"
-                exec printf('luafile %s', rc)
-            else
-                exec printf('source %s', rc)
+        for fname in s:fnames
```

Comment:
Thanks for doing this. I added in the lua support as a quick feature to keep working on something else that required it, but the old implementation would ignore `.vimrc.lua` if `.vimrc` existed in the same location.